### PR TITLE
Datasource: Make sure data proxy timeout applies to HTTP client

### LINKED ISSUE #25863: Azure log analytics query timeout after 30 seconds even though timeout is configured to 60 seconds
**What happened**:
`Get "https://api.loganalytics.io/v1/workspaces/...": net/http: request canceled (Client.Timeout exceeded while awaiting headers)`.

**What you expected to happen**:
Having [timeout](https://grafana.com/docs/grafana/latest/administration/configuration/#timeout) configured to 60 seconds I would expect request to timeout after 60 seconds.

**Environment**:
- Grafana version: 7.0.4
- Data source type & version: Azure Monitor / log analytics

### PR DESCRIPTION
**Pull request by marefr**
**What this PR does / why we need it**
For backend data sources executing in the backend (not through data proxy) make sure that the timeout applies to cached HTTP client.

**Which issue(s) this PR fixes**:
Fixes #25863 

**Special notes for your reviewer**:
Don't think the timeout setting really works for data proxy, but that's another problem/issue.

### CODE REVIEW

**Review by kylebrandt — APPROVED** [review]
:rocket:

File: conf/defaults.ini [hunk 1]
```diff
@@ -126,7 +126,8 @@ connstr =
 # This enables data proxy logging, default is false
 logging = false
 
-# How long the data proxy should wait before timing out default is 30 (seconds)
+# How long the data proxy should wait before timing out default is 30 (seconds).
```
Comment (oddlittlebird) [inline conf/defaults.ini:129]:
```suggestion
# How long the data proxy waits before timing out, default is 30 seconds.
```

File: conf/defaults.ini [hunk 2]
```diff
@@ -126,7 +126,8 @@ connstr =
...
+# Note: This setting also applies to core backend HTTP data sources where queries are requested using an HTTP client with timeout set.
```
Comment (oddlittlebird) [inline conf/defaults.ini:130]:
```suggestion
# This setting also applies to core backend HTTP data sources where queries requests use an HTTP client with timeout set.
```

Comment (oddlittlebird) [reply conf/defaults.ini:130]: If you accept this suggestion, please apply everywhere it is used.

File: docs/sources/administration/configuration.md [hunk 3]
```diff
@@ -339,7 +339,9 @@ This enables data proxy logging, default is `false`.
 
 ### timeout
 
-How long the data proxy should wait before timing out. Default is `30` (seconds)
+How long the data proxy should wait before timing out. Default is `30` (seconds).
```
Comment (oddlittlebird) [inline docs/sources/administration/configuration.md:342]:
```suggestion
How long the data proxy should wait before timing out. Default is 30 seconds.
```

File: docs/sources/administration/configuration.md [hunk 4]
```diff
@@ -537,7 +539,7 @@ Set the default UI theme: `dark` or `light`. Default is `dark`.
 
 ### External user management
 
-If you manage users externally you can replace the user invite button for organizations with a link to an external site together with a description. 
+If you manage users externally you can replace the user invite button for organizations with a link to an external site together with a description.
```
Comment (oddlittlebird) [inline docs/sources/administration/configuration.md:542]:
```suggestion
If you manage users externally, then you can replace the user invite button for organizations with a link to an external site together with a customized description.
```

**Review by oddlittlebird — APPROVED** [review]
Minor editing suggestions.

File: pkg/models/datasource_cache.go [hunk 5]
```diff
@@ -51,7 +51,7 @@ func (ds *DataSource) GetHttpClient() (*http.Client, error) {
 	}
 
 	return &http.Client{
-		Timeout:   30 * time.Second,
+		Timeout:   time.Duration(setting.DataProxyTimeout) * time.Second,
```
Comment (bergquist) [inline pkg/models/datasource_cache.go:54]: Im... amazed that we didn't change this before? I know we tried but I guess we updated the code in the wrong place? I'm quite puzzled. Great Work! :)

**Review by bergquist — APPROVED** [review]
LGTM

File: conf/defaults.ini (hunk 2 above)
Comment (marefr) [reply conf/defaults.ini:130]:
```suggestion
# This setting also applies to core backend HTTP data sources where query requests use an HTTP client with timeout set.
```

File: conf/sample.ini [hunk 6]
```diff
@@ -128,6 +128,7 @@
 ;logging = false
 
 # How long the data proxy should wait before timing out default is 30 (seconds)
+# Note: This setting also applies to core backend HTTP data sources where queries are requested using an HTTP client with timeout set.
```
Comment (marefr) [inline conf/sample.ini:131]:
```suggestion
# This setting also applies to core backend HTTP data sources where queries requests use an HTTP client with timeout set.
```

File: conf/sample.ini [hunk 7]
```diff
@@ -128,6 +128,7 @@
 ;logging = false
 
 # How long the data proxy should wait before timing out default is 30 (seconds)
```
Comment (marefr) [inline conf/sample.ini:130]:
```suggestion
# How long the data proxy waits before timing out, default is 30 seconds.
```

File: docs/sources/administration/configuration.md (hunk 4 above)
Comment (marefr) [reply docs/sources/administration/configuration.md:542]: Please handle this in a separate PR

File: docs/sources/administration/configuration.md [hunk 8]
```diff
@@ -339,7 +339,9 @@ This enables data proxy logging, default is `false`.
 
 ### timeout
 
-How long the data proxy should wait before timing out. Default is `30` (seconds)
+How long the data proxy should wait before timing out. Default is `30` (seconds).
+
+> **Note:** This setting also applies to core backend HTTP data sources where queries are requested using an HTTP client with timeout set.
```
Comment (marefr) [inline docs/sources/administration/configuration.md:344]:
```suggestion
This setting also applies to core backend HTTP data sources where query requests use an HTTP client with timeout set.
```

**Pull request merged**